lib/slimmer/processors/nonce_inserter.rb

```diff
@@ -0,0 +1,20 @@
+module Slimmer::Processors
+  class NonceInserter
+    def initialize(env)
+      # As Rails is an optional dependency of this gem quietly do nothing if Rails
+      # classes don't exist.
+      @nonce = if defined?(ActionDispatch::Request)
+                 ActionDispatch::Request.new(env).content_security_policy_nonce
+               end
+    end
+
+    def filter(_src, dest)
+      return unless @nonce
+
+      # Add the nonce attribute to script elements that don't have a src attribute
+      dest.css("script:not([src])").each do |script|
```

Comment:
Could you add something to either to the comment or the commit message about why we only do it for scripts without src?

Reply:
Yes, good shout.

Reply:
I've added some information to both commit and comment.